src/d3a/d3a_core/export.py

```diff
@@ -215,10 +215,13 @@ def _export_area_clearing_rate(self, area, directory, file_suffix, is_first):
                 if is_first:
                     writer.writerow(labels)
                 for market in area.past_markets:
-                    for time, clearing in (
-                            bid_offer_matcher.matcher.match_algorithm.state.clearing.items()):
+                    market_clearing = bid_offer_matcher.matcher.match_algorithm.state.clearing.get(
```

Comment:
Nit: please be aware that `bid_offer_matcher.matcher.match_algorithm` only works for one of the two matches (the `MycoInternalMatcher`), so this is not a "stable" interface (in the sense that we can't always call this method, because it will break if we change the matcher class). Anyway, this is not to be dealt with in this PR.

Reply:
This method is only called when logical condition satisfied which targets PAC market is satisfied

`
if ConstSettings.IAASettings.MARKET_TYPE == 2 and ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE == BidOfferMatchAlgoEnum.PAY_AS_CLEAR.value:
`

Reply:
Yes, but I'd say that this is not what interfaces are for. We should be able to use all the interface methods in all the classes that implement the interface, without needing `if` statements and without breaking stuff during the call. (The null-object pattern also relates to this IMO, in case you'd like to check it out).

Reply:
I will leave for future refactoring being targeted as part of [D3ASIM-3600](https://gridsingularity.atlassian.net/browse/D3ASIM-3600)